app/src/androidTest/java/org/kiwix/kiwixmobile/main/TopLevelDestinationRobot.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.main
+
+import applyWithViewHierarchyPrinting
+import org.kiwix.kiwixmobile.BaseRobot
+import org.kiwix.kiwixmobile.Findable.StringId.TextId
+import org.kiwix.kiwixmobile.Findable.ViewId
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.help.HelpRobot
+import org.kiwix.kiwixmobile.help.help
+import org.kiwix.kiwixmobile.nav.destination.library.LibraryRobot
+import org.kiwix.kiwixmobile.nav.destination.library.OnlineLibraryRobot
+import org.kiwix.kiwixmobile.nav.destination.library.library
+import org.kiwix.kiwixmobile.nav.destination.library.onlineLibrary
+import org.kiwix.kiwixmobile.nav.destination.reader.ReaderRobot
+import org.kiwix.kiwixmobile.nav.destination.reader.reader
+import org.kiwix.kiwixmobile.page.bookmarks.BookmarksRobot
+import org.kiwix.kiwixmobile.page.bookmarks.bookmarks
+import org.kiwix.kiwixmobile.page.history.HistoryRobot
+import org.kiwix.kiwixmobile.page.history.history
+import org.kiwix.kiwixmobile.settings.SettingsRobot
+import org.kiwix.kiwixmobile.settings.settingsRobo
+import org.kiwix.kiwixmobile.utils.StandardActions.openDrawer
+import org.kiwix.kiwixmobile.webserver.ZimHostRobot
+import org.kiwix.kiwixmobile.webserver.zimHost
+
+fun topLevel(func: TopLevelDestinationRobot.() -> Unit) =
+  TopLevelDestinationRobot().applyWithViewHierarchyPrinting(func)
+
+class TopLevelDestinationRobot : BaseRobot() {
+
+  init {
+    isVisible(ViewId(R.id.navigation_container))
+  }
+
+  fun clickReaderOnBottomNav(func: ReaderRobot.() -> Unit) {
+    clickOn(ViewId(R.id.readerFragment))
+    reader(func)
+  }
+
+  fun clickLibraryOnBottomNav(func: LibraryRobot.() -> Unit) {
+    clickOn(ViewId(R.id.libraryFragment))
+    library(func)
+    pressBack()
+  }
+
+  fun clickDownloadOnBottomNav(func: OnlineLibraryRobot.() -> Unit) {
+    clickOn(ViewId(R.id.downloadsFragment))
+    onlineLibrary(func)
+    pressBack()
+  }
+
+  private fun navDrawer(func: NavDrawerRobot.() -> Unit) =
+    NavDrawerRobot().applyWithViewHierarchyPrinting(func)
+
+  inner class NavDrawerRobot : BaseRobot() {
```

Comment:
this isn't really taking advantage of being a robot because the only function is the amorphous `inNavDrawer(navDrawerAction: () -> Unit)` which just accepts any old thing and we want to use receiver functions to limit the scope and make a nice dsl.

`clickBookmarksOnNavDrawer(func: BookmarksRobot.() -> Unit)` would want to look like
```
clickOnNavDrawer{
 clickOnBookmarks(func)
}
```

also NavDrawerRobot should be private

Reply:
This really would become too much overkill, writing too much just for a method that is being used only at a few places, despite the fact that the overkill will make the code look _slightly_ better only, no big effects. Is this really needed?

Reply:
That is what I was saying!
>Yeah the current scenario is fine, just the other scenario is better because we have Typed the problem space, reducing potential misuse but also comes with a larger overhead in time/cognition.
 

Overkill but if you are going to write a robot then it should be written correctly with the ui elements of that robot only accessible from that robot's scope. I am happy with the previous state of the code if you want to revert a commit and we can hopefully push this over the line today

Reply:
Cool